security/src/main/java/org/apache/geode/examples/security/Example.java

```diff
@@ -0,0 +1,156 @@
+          + "Ernest Hemingway,F. Scott Fitzgerald,Henry David Thoreau,Henry Wadsworth Longfellow,"
+          + "Herman Melville,Jean-Paul Sartre,Mark Twain,Orson Scott Card,Ray Bradbury,Robert Benchley,"
+          + "Somerset Maugham,Stephen King,Terry Pratchett,Ursula K. Le Guin,William Faulkner")
+              .split(",");
+
+  private String username;
+  private String password;
+  private final ClientCache cache;
+
+
+  private Example(String username, String password) {
+    this.username = username;
+    this.password = password;
+    Properties props = new Properties();
+    props.setProperty(ExampleAuthInit.USER_NAME, this.username);
+    props.setProperty(ExampleAuthInit.PASSWORD, this.password);
+    props.setProperty(SECURITY_CLIENT_AUTH_INIT, ExampleAuthInit.class.getName());
+    // connect to the locator using default port 10334
+    cache = new ClientCacheFactory(props).setPoolSubscriptionEnabled(true)
+        .addPoolLocator("localhost", 10334).create();
+  }
+
+  /** A far-from-comprehensive example of integrated security in action. */
+  public static void main(String[] args) {
+    printCurrentData();
+
+    getDataExample("dataReader", false);
+    getDataExample("dataWriter", true);
+
+    copyDataExample("dataWriter", true);
+    copyDataExample("dataReader", true);
+    copyDataExample("dataAdmin", false);
+
+    printCurrentData();
+  }
+
+  /** Prints the data currently present in both /region1 and /region2 */
+  private static void printCurrentData() {
+    Example example = new Example("super-user", "123");
+    Region<String, String> region1 =
+        example.cache.<String, String>createClientRegionFactory(ClientRegionShortcut.CACHING_PROXY)
+            .create(REGION1);
+    logger.info("======\nContents of /region1:");
+    example.getData(region1, false);
+    Region<String, String> region2 =
+        example.cache.<String, String>createClientRegionFactory(ClientRegionShortcut.CACHING_PROXY)
+            .create(REGION2);
+    logger.info("======\nContents of /region2:");
+    example.getData(region2, false);
+
+    example.cache.close(true);
+  }
+
+
+  /**
+   * Attempt to fetch and print the data in /region1 using the authorization of the
+   * provided @username.
+   */
+  private static void getDataExample(String username, boolean expectAuthFailure) {
+    logger.info(String.format("Starting get example with user '%s', expecting %s", username,
+        expectAuthFailure ? "authorization failure." : "success."));
+
+    Example example = new Example(username, "123");
+
+    Region<String, String> region =
+        example.cache.<String, String>createClientRegionFactory(ClientRegionShortcut.CACHING_PROXY)
+            .create(REGION1);
+
+    example.getData(region, expectAuthFailure);
+
+    example.cache.close(true);
+  }
+
+  /** Attempt to print the data present in the given @region. */
+  private void getData(Region<String, String> region, boolean expectAuthFailure) {
```

Comment:
Can we just always log the auth failures in this example and get rid of the whole `expectAuthFailure` flag?